### DISCUSSION TITLE
Sahi work on non axis oriented bounding box? Like yolo obb?

### DISCUSSION THREAD

**Discussion by karthikdatta98** [obss/sahi#987] [Q&A] [answered]
Does sahi work on non axis oriented bounding box  models?

**Comment by whao159107** [comment]
Why is there no answer or example? Thank you.

**Reply by fcakyon** [reply]
Hey @whao159107, Ultralytics YOLOv5U/YOLOv8/YOLO11 OBB models are supported with latest sahi release: https://github.com/obss/sahi/releases/tag/0.11.20

Usage is similar to Ultralytics det/seg models: https://github.com/obss/sahi/blob/main/demo/inference_for_ultralytics.ipynb

**Comment by robmarkcole** [comment]
The answere is no, and trying to do so will fail with 
```
prediction_result = [result.boxes.data for result in prediction_result]

AttributeError: 'NoneType' object has no attribute 'data'
```
This is because OBB are returned at `result.obb`. I also believe the sahi algo does not currently support OBB.
However the obb results do contain regular boxes too (`xyxy`), so it should be feasible to use those:
```
ultralytics.engine.results.OBB object with attributes:

cls: tensor([5.])
conf: tensor([0.2714])
data: tensor([[1.7813e+02, 1.3286e+02, 4.0502e+02, 3.2889e+02, 2.5792e+00, 2.7138e-01, 5.0000e+00]])
id: None
is_track: False
orig_shape: (1024, 1024)
shape: torch.Size([1, 7])
xywhr: tensor([[178.1333, 132.8626, 405.0189, 328.8881,   2.5792]])
xyxy: tensor([[ -80.8699, -114.2355,  437.1364,  379.9607]])
xyxyxyxy: tensor([[[ -80.8699,  101.7294],
         [  94.5004,  379.9607],
         [ 437.1364,  163.9958],
         [ 261.7661, -114.2355]]])
xyxyxyxyn: tensor([[[-0.0790,  0.0993],
         [ 0.0923,  0.3711],
         [ 0.4269,  0.1602],
         [ 0.2556, -0.1116]]])
```

**Reply by fcakyon** [reply]
Hey @robmarkcole , Ultralytics YOLOv5U/YOLOv8/YOLO11 OBB models are supported with latest sahi release: https://github.com/obss/sahi/releases/tag/0.11.20

Usage is similar to Ultralytics det/seg models: https://github.com/obss/sahi/blob/main/demo/inference_for_ultralytics.ipynb

**Comment by ZSBSB** [comment]
这样做应该是不行的

**Reply by fcakyon** [reply]
Hey @ZSBSB, Ultralytics YOLOv5U/YOLOv8/YOLO11 OBB models are supported with latest sahi release: https://github.com/obss/sahi/releases/tag/0.11.20

Usage is similar to Ultralytics det/seg models: https://github.com/obss/sahi/blob/main/demo/inference_for_ultralytics.ipynb

**Comment by fcakyon** [comment, marked as answer]
Yes, Ultralytics YOLOv5U/YOLOv8/YOLO11 OBB models are supported with latest sahi release: https://github.com/obss/sahi/releases/tag/0.11.20

Usage is similar to Ultralytics det/seg models: https://github.com/obss/sahi/blob/main/demo/inference_for_ultralytics.ipynb

**Comment by xhmelody** [comment]
You are very welcome! I am really eager to try this new change.




M
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "obss/sahi"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年12月17日(星期二) 凌晨2:06
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [obss/sahi] Sahi work on non axis oriented bounding box? Like yolo obb? (Discussion #987)





 
Yes, Ultralytics YOLOv5U/YOLOv8/YOLO11 OBB models are supported with latest sahi release: https://github.com/obss/sahi/releases/tag/0.11.20
 
Usage is similar to Ultralytics det/seg models: https://github.com/obss/sahi/blob/main/demo/inference_for_ultralytics.ipynb
